### ISSUE TITLE
Upgrade to Dagger 2

### ISSUE THREAD

**Issue by joielechong** [spirosoik/RapidAndroidFramework#39]
Any chance that you upgrading this project to dagger 2?..
I just recently starting on using dagger. I'm trying to upgrade dagger by myself, but it starting to eat my hair :(

**Comment by spirosoik** [comment]
@joielechong Yes of course. These days I will create a new repository with different approach. Check this issue https://github.com/spirosoik/RapidAndroidFramework/issues/35 .

If you want you can use the repos bellow also which for now missing the JEDI version (more layers + uncle's bob architecture + more utils as RapidAndroidFramework) which will be the new repo as described above.
https://github.com/spirosoik/AndroidArchitectureJedis -> (Coming soon - Advanced) 
https://github.com/spirosoik/AndroidArchitecturePadawans -> For newcomers 
https://github.com/spirosoik/AndroidArchitecture-BabyPadawans

**Comment by joielechong** [comment]
I really like the Uncle's Bob Clean Architecture approach, I'll wait for the coming soon https://github.com/spirosoik/AndroidArchitectureJedis 

For now I learn as BabyPadawans, though I don't think it really means to a baby..:D :P

**Comment by spirosoik** [comment]
@joielechong hahahaha yes I did a presentation with these to newcomers and they were completely lost but after the second presentation, they understood a few things. this uses dagger2 https://github.com/spirosoik/AndroidArchitecturePadawans I think this is for you man

**Comment by spirosoik** [comment]
@joielechong Also for Data-mappers you can use http://modelmapper.org/  or  http://mapstruct.org/ but for this repo and for learning purposes I wrote the Data-mappers in order to understand the logic.

**Comment by joielechong** [comment]
@spirosoik I really know their pain, cause I also lost when trying to know about using specific architecture. But the pain really worth the gain.
I guess the main problem is the third party library like retrofit and dagger. They "obsfucate" the architecture.
Btw, Could you send me your presentation material?

Though I never heard about those data-mappers, I'll try to learn it. Thank you for the information.

**Comment by spirosoik** [comment]
@joielechong I think that retrofit and dagger helps and they adapt nicely into Uncle Bob's clean architecture - maybe obsfucate a little bit -. If you read in many articles and you will understand it during the development that there are so many 3rd party libraries that can help you and it's a good practice to "not reinvent the wheel". Of course for learning purposes  you might keep it absolutely simple ( I did the mistake as presenter read [here!](https://medium.com/@spirosoik/the-result-of-presenter-s-mistakes-4b4f624603f2)

The presentations are too generic but you can find them [here!](https://slides.com/spirosoikonomakis)

**Comment by joielechong** [comment]
@spirosoik you're correct, those library really help when applying Uncle Bob's clean architecture. In my case, I completely lost because haven't learn about those library first.

Thank you for the slide. But I must turn off AdGuard, a chrome extension, because it reports: "This web page at slides.com, has been reported as a phishing page and has been blocked based on your security preferences."

**Comment by spirosoik** [comment]
@joielechong for me works with AdGuard. this site for slides/presentations it's popular.

**Comment by joielechong** [comment]
Your article really hit the nails. Inspiring article by the way..:).

**Comment by spirosoik** [comment]
@joielechong thanks, I thought that it was a good idea to write for this big fail :)
